content/en/docs/Writing policies/external-data-sources.md

```diff
@@ -0,0 +1,440 @@
+          "resourceVersion": "155",
+          "creationTimestamp": "2021-01-19T20:20:37Z",
+          "managedFields": [
+            {
+              "manager": "kube-apiserver",
+              "operation": "Update",
+              "apiVersion": "v1",
+              "time": "2021-01-19T20:20:37Z",
+              "fieldsType": "FieldsV1",
+              "fieldsV1": {
+                "f:status": {
+                  "f:phase": {}
+                }
+              }
+            }
+          ]
+        },
+        "spec": {
+          "finalizers": [
+            "kubernetes"
+          ]
+        },
+        "status": {
+          "phase": "Active"
+        }
+      },
+
+      ...
+
+```
+
+To process this data in JMESPath, reference the `items`. Here is an example, to extract a few metadata fields across all namespaces:
+
+```sh
+kubectl get --raw /api/v1/namespaces | jp "items[*].{name: metadata.name, creationTime: metadata.creationTimestamp}"
+```
+
+This produces a new JSON list of objects with properties `name` and `creationTime`. 
+
+```json
+[
+  {
+    "creationTimestamp": "2021-01-19T20:20:37Z",
+    "name": "default"
+  },
+  {
+    "creationTimestamp": "2021-01-19T20:20:36Z",
+    "name": "kube-node-lease"
+  },
+
+  ...
+
+```
+
+To find an item in the list you can use JMESPath filters. For example, this command will match a namespace by its name:
+
+```sh
+ kubectl get --raw /api/v1/namespaces | jp "items[?metadata.name == 'default'].{uid: metadata.uid, creationTimestamp: metadata.creationTimestamp}"
+```
+
+In addition to wildcards and filters, JMESPath has many additional powerful features including several useful functions. Be sure to go through the [JMESPath tutorial](https://jmespath.org/tutorial.html) and try the interactive examples.
+
+
+### Sample Policy: Limit Services of type LoadBalancer in a namespace 
+
+Here is a complete sample policy that limits each namespace to a single service of type `LoadBalancer`. 
+
+
+```yaml
+apiVersion : kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: limits
+spec:
+  validationFailureAction: enforce
+  rules:
+  - name: limit-lb-svc
+    match:
+      resources:
+        kinds:
+        - Service
+    context:
+    - name: serviceCount
+      apiCall:
+        urlPath: "/api/v1/namespaces/{{ request.object.metadata.namespace }}/services"
+        jmesPath: "items[?spec.type == 'LoadBalancer'] | length(@)"    
+    preconditions:
+    - key: "{{ request.operation }}"
+      operator: Equals
+      value: "CREATE"
+    validate:
+      message: "Only one LoadBalancer service is allowed per namespace"
+      deny:
+        conditions:
+        - key: "{{ serviceCount }}"
+          operator: GreaterThanOrEquals
+          value: 1
+
+```
+
+It retrieves the list of services in the namespace and stores the count of services of ype `LoadBalancer` in a variable called `serviceCount`. A `deny` rule is used to ensure that the count cannot exceed 1.
```

Comment:
"This sample policy retrieves the list of Services in the Namespace and stores the count of type `LoadBalancer` in a variable called `serviceCount`."